kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java

```diff
@@ -142,21 +146,22 @@ protected void doInit() {
             System.exit(1);
           });
       this.shutdownDuration = getShutdownDuration();
+      this.initialDelay = InitialDelayConfigProvider.getInstance().getInitialDelay(streamsConfig);
+      this.isSleeping = false;
       getLogger().info("kafka streams topologies: {}", topology.describe());
     } catch (Exception e) {
       getLogger().error("Error initializing - ", e);
-      e.printStackTrace();
       System.exit(1);
     }
   }
 
   @Override
   protected void doStart() {
     try {
+      delayStartup(initialDelay.toMillis());
```

Comment:
no need to pass any argument here, since initialDelay is already a member variable.

Reply:
looks cleaner this way